projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -775,12 +775,28 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
         return this.columnList.some((col) => col.groupable && !col.columnGroup);
     }
 
+    /**
+     * Returns whether the `IgxGridComponent` has group area.
+     *
+     * @example
+     * ```typescript
+     * let isGroupAreaVisible = this.grid.showGroupArea;
+     * ```
+     *
+     * @example
+     * ```html
+     * <igx-grid #grid [data]="Data" [showGroupArea]="false"></igx-grid>
+     * ```
+     */
     @Input()
     public get showGroupArea(): boolean {
         return this._showGroupArea;
     }
     public set showGroupArea(value: boolean) {
         this._showGroupArea = value;
+        if (!this._init) {
+            this.calculateGridSizes();
+        }
```

Comment:
```suggestion
        this.notifyChanges(true);
```
Let's defer the calculation for the next cycle not in the middle of the setter